### PR TITLE
PR #4: Backend of subfolder, dynamic creation of teams/products, algolia dependency removed completely, frontend enhanced

### PR DESCRIPTION
**Pull request by yashcode00**
🚀 **PR 4: Summary:**

The following are the things done in this PR:

1. Modifies modified Get APIs for the products.go, which now fetches the data from the database instead of Algolia.
2. Added "Post" method for the products.go API, o incorporate the dynamic addition of new products using frontend POST requests with proper validation.
3. Similarly, made Get and Post API handlers for the newly added subfolder namely "Teams" with almost the same functionality.
4. Made the model for teams so as to make its separate table in the Postgres database so as to dynamically operate similar to the products.
5. Made all functions for upserting, get for teams table, along with making its associations with the BU tables like the foreign key (referential integrity constraint).
6. Similarly, added the associations in the product.go (Model for the product).
7. Now, modified the frontend controller for the dashboard to incorporate the button requests of "submit the form" and "creating the form" both aimed to facilitate the dynamic addition of new products and teams on demand.
8. Created a new ember-cli-modal-dailogue for showing the form when creating a new product/team us clicked.]
9. Made necessary backend connections.
10.  Made both dropdowns of team and product when we create new doc dependent, added filter-options function for teams to filter based upon the selected product (BU).

> [cherry-picked from  [0d8ba7a](https://github.com/razorpay/hermes/pull/5/commits/0d8ba7ac6f82f07c641b7c714252520bab6452b1) , [e5e71d6](https://github.com/razorpay/hermes/pull/5/commits/e5e71d63a1157001b424054776db4a13178ce004) ]

Thank you! 😊

### CODE REVIEW

File: internal/api/products.go [hunk 1]
```diff
@@ -2,58 +2,210 @@ package api
...
+		}
+	}
+
+	return productData, nil
+}
+
+// AddNewProducts This helper fuction add the newly added product in both algolia and upserts it
+// in the postgres Database
+func AddNewProducts(ar *algolia.Client,
+	aw *algolia.Client, db *gorm.DB, req ProductRequest) error {
```
Comment (ashisht99x) [inline internal/api/products.go:120]: Please remove commented code

File: internal/api/products.go [hunk 2]
```diff
@@ -2,58 +2,210 @@ package api
...
-// data from Algolia
-func getProductsData(a *algolia.Client) (map[string]structs.ProductData, error) {
-	p := structs.Products{
-		ObjectID: "products",
-		Data:     make(map[string]structs.ProductData, 0),
-	}
+// data from Database
+func getProductsData(db *gorm.DB) (map[string]struct {
+	Abbreviation   string      `json:"abbreviation"`
+	PerDocTypeData interface{} `json:"perDocTypeData"`
```
Comment (ashisht99x) [inline internal/api/products.go:91]: "PerDocTypeData" this node name is not revealing the puspose. Please find a better name

**Review by ashisht99x — APPROVED** [review]

File: internal/api/products.go (hunk 2 above)
Comment (yashcode00) [reply internal/api/products.go:91]: this attribute is never used and normally initialized to null, so the purpose is not quite clear at this point of time.

File: internal/api/products.go (hunk 1 above)
Comment (yashcode00) [reply internal/api/products.go:120]: done

**Pull request merged**